client_documents_test.go

```diff
@@ -166,17 +164,16 @@ func TestClientDocuments_List(t *testing.T) {
 
 	var list []docTest
 	err = client.Documents(indexUID).List(ListDocumentsRequest{
-		Offset: 0,
-		Limit:  100,
+		Offset: 1,
+		Limit:  1,
 	}, &list)
 
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	// tests are running in parallel so there can be more than 1 docs
-	if len(list) < 1 {
-		t.Fatal("number of doc should be at least 1")
+	if list[0].ID != "456" {
```

Comment:
```suggestion
	if len(list) == 0 || list[0].ID != "456" {
```

Reply:
This test actually replaced the old one in your PR, which was quite simple but still seems pertinent. Also, an error in retrieving the list would cause a crash instead of just failing the test. So I suggest adding this simple check before trying to access the slice first element, inb case it doesn't exist. We have the double check and avoid a crash

Reply:
Good idea including that safety in the test 👍.